File: Array/leetcode/easy/SingleNumber.py
#https://leetcode.com/problems/single-number/
def single_number(arr, n):
    result = 0
    for value in arr:
        result ^= value
    return result        

arr1 = [1, 2, 1, 2, 4]
print("Test Case 1:", single_number(arr1, len(arr1)))

arr2 = [2, 2, 1]
print("Test Case 2:", single_number(arr2, len(arr2)))

arr3 = [4, 1, 2, 1, 2]
print("Test Case 3:", single_number(arr3, len(arr3)))

File: Array/leetcode/easy/SingleNumber.go
// https://leetcode.com/problems/single-number/
package main

import "fmt"

func singleNumber(nums []int, n int) int {
	result := 0

	for _, value := range(nums) {
	   result ^= value
	}

	return result
}

func singleNumberMain() {
    arr1 := []int{1, 2, 1, 2, 4}
    fmt.Println("Test Case 1:", singleNumber(arr1, len(arr1)))

    arr2 := []int{2, 2, 1}
    fmt.Println("Test Case 2:", singleNumber(arr2, len(arr2)))

    arr3 := []int{4, 1, 2, 1, 2}
    fmt.Println("Test Case 3:", singleNumber(arr3, len(arr3)))
}
